toolchain/check/testdata/interop/cpp/function/operators.carbon

```diff
@@ -12,31 +12,102 @@
 // TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/interop/cpp/function/operators.carbon
 
 // ============================================================================
-// Negate
+// Unary operators
 // ============================================================================
 
-// --- negate.h
+// --- unary_operators.h
 
 class C {};
+
+// Increment and Decrement.
+// TODO: Change `operand` and return value to reference when it is supported.
+auto operator++(C operand) -> C;
+auto operator--(C operand) -> C;
+
+// Arithmetic.
 auto operator-(C operand) -> C;
 
-// --- fail_todo_import_negate.carbon
+// --- import_unary_operators.carbon
 
 library "[[@TEST_NAME]]";
 
-import Cpp library "negate.h";
+import Cpp library "unary_operators.h";
 
 fn F() {
   //@dump-sem-ir-begin
-  let c1: Cpp.C = Cpp.C.C();
-  // CHECK:STDERR: fail_todo_import_negate.carbon:[[@LINE+4]]:19: error: cannot access member of interface `Core.Negate` in type `Cpp.C` that does not implement that interface [MissingImplInMemberAccess]
-  // CHECK:STDERR:   let c2: Cpp.C = -c1;
-  // CHECK:STDERR:                   ^~~
-  // CHECK:STDERR:
-  let c2: Cpp.C = -c1;
+  let c: Cpp.C = Cpp.C.C();
+
+  // Increment and Decrement.
+  ++c;
+  --c;
+
+  // Arithmetic.
+  let minus: Cpp.C = -c;
   //@dump-sem-ir-end
 }
 
+// ============================================================================
+// Prefix increment and decrement vs postfix increment and decrement
+// ============================================================================
+
+// --- prefix_inc_and_dec.h
+
+class Prefix {};
+auto operator++(Prefix operand) -> Prefix;
+auto operator--(Prefix operand) -> Prefix;
+
+// --- fail_prefix_calling_postfix.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "prefix_inc_and_dec.h";
+
+fn F() {
+  let prefix: Cpp.Prefix = Cpp.Prefix.Prefix();
+  // CHECK:STDERR: fail_prefix_calling_postfix.carbon:[[@LINE+8]]:9: error: expected `;` after expression statement [ExpectedExprSemi]
+  // CHECK:STDERR:   prefix++;
+  // CHECK:STDERR:         ^~
+  // CHECK:STDERR:
+  // CHECK:STDERR: fail_prefix_calling_postfix.carbon:[[@LINE+4]]:3: error: semantics TODO: `handle invalid parse trees in `check`` [SemanticsTodo]
+  // CHECK:STDERR:   prefix++;
+  // CHECK:STDERR:   ^~~~~~~~~
+  // CHECK:STDERR:
+  prefix++;
+  // CHECK:STDERR: fail_prefix_calling_postfix.carbon:[[@LINE+4]]:9: error: expected `;` after expression statement [ExpectedExprSemi]
+  // CHECK:STDERR:   prefix--;
+  // CHECK:STDERR:         ^~
+  // CHECK:STDERR:
+  prefix--;
+}
```

Comment:
This test looks like its also covered by the `fail_todo_import_unsupported_unary_operators` so I think it could go away. The one below is different though, and we should keep that one.

Reply:
Thanks for clarifying.
We probably want to be able to (eventually) call C++ postfix operators.
If we do, I don't know what the syntax would be but it makes sense to have this test (and update the syntax if we decide to do it differently).
If we decide to disallow calling the C++ postfix operators, we should probably remove this test.